sprockets/http/app.py

```diff
@@ -40,13 +45,18 @@ def on_shutdown_ready(self):
 
     def _maybe_stop(self):
         now = self.io_loop.time()
-        if (now < self.__deadline and
-                (self.io_loop._callbacks or self.io_loop._timeouts)):
+        if now < self.__deadline and self._ioloop_has_tasks():
             self.io_loop.add_timeout(now + 1, self._maybe_stop)
         else:
             self.io_loop.stop()
             self.logger.info('stopped IOLoop')
 
+    def _ioloop_has_tasks(self):
+        if ASYNCIO_TORNADO:
+            return bool(asyncio.Task.all_tasks())
```

Comment:
Question: Can there be an async task outside of Tornado's ioloop? Is it possible? Likely? If it is possible but not likely, then this is good for a first cut. It can always improve later.

Reply:
Yes - it's possible. Likely for our use? No.